core/services/ardupilot_manager/typedefs.py

```diff
@@ -100,12 +103,31 @@ class FirmwareFormat(str, Enum):
     ELF = "ELF"
 
 
-class FlightControllerType(IntEnum):
-    """Supported flight-controller types."""
+class PlatformType(str, Enum):
+    Serial = "serial"
+    Linux = "linux"
+    SITL = "sitl"
+    Unknown = "unknown"
+
+
+def get_platform_type(platform: Platform) -> PlatformType:
+    platform_types = {
+        Platform.Pixhawk1: PlatformType.Serial,
+        Platform.NavigatorR3: PlatformType.Linux,
+        Platform.NavigatorR4: PlatformType.Linux,
+        Platform.NavigatorR5: PlatformType.Linux,
+        Platform.SITL: PlatformType.SITL,
+    }
+    return platform_types.get(platform, PlatformType.Unknown)
```

Comment:
You can move the function to be inside the enum class, this will make it more clear to use. 
```python
class Platform(str, Enum):
    """Valid Ardupilot platform types.
    The Enum values are 1:1 representations of the platforms available on the ArduPilot manifest."""

    Pixhawk1 = "Pixhawk1"
    NavigatorR3 = "navigator"
    NavigatorR4 = "navigator"
    NavigatorR5 = "navigator"
    #SITL = get_sitl_platform_name(machine())
    Undefined = "Undefined"
    
    def type(self) -> PlatformType:
        platform_types = {
            Platform.Pixhawk1: PlatformType.Serial,
            Platform.NavigatorR3: PlatformType.Linux,
            Platform.NavigatorR4: PlatformType.Linux,
            Platform.NavigatorR5: PlatformType.Linux,
        }
        return platform_types.get(self.value, PlatformType.Unknown)
    
print(Platform.Pixhawk1.type())
```

Reply:
Good idea